client/utils/rest.go

```diff
@@ -93,57 +92,10 @@ func ParseFloat64OrReturnBadRequest(w http.ResponseWriter, s string, defaultIfEm
 //-----------------------------------------------------------------------------
 // Building / Sending utilities
 
-// BaseReq defines a structure that can be embedded in other request structures
-// that all share common "base" fields.
-type BaseReq struct {
-	From          string       `json:"from"`
-	Password      string       `json:"password"`
-	Memo          string       `json:"memo"`
-	ChainID       string       `json:"chain_id"`
-	AccountNumber uint64       `json:"account_number"`
-	Sequence      uint64       `json:"sequence"`
-	Fees          sdk.Coins    `json:"fees"`
-	GasPrices     sdk.DecCoins `json:"gas_prices"`
-	Gas           string       `json:"gas"`
-	GasAdjustment string       `json:"gas_adjustment"`
-	GenerateOnly  bool         `json:"generate_only"`
-	Simulate      bool         `json:"simulate"`
-}
-
-// NewBaseReq creates a new basic request instance and sanitizes its values
-func NewBaseReq(
-	from, password, memo, chainID string, gas, gasAdjustment string,
-	accNumber, seq uint64, fees sdk.Coins, gasPrices sdk.DecCoins, genOnly, simulate bool,
-) BaseReq {
-
-	return BaseReq{
-		From:          strings.TrimSpace(from),
-		Password:      password,
-		Memo:          strings.TrimSpace(memo),
-		ChainID:       strings.TrimSpace(chainID),
-		Fees:          fees,
-		GasPrices:     gasPrices,
-		Gas:           strings.TrimSpace(gas),
-		GasAdjustment: strings.TrimSpace(gasAdjustment),
-		AccountNumber: accNumber,
-		Sequence:      seq,
-		GenerateOnly:  genOnly,
-		Simulate:      simulate,
-	}
-}
-
-// Sanitize performs basic sanitization on a BaseReq object.
-func (br BaseReq) Sanitize() BaseReq {
-	return NewBaseReq(
-		br.From, br.Password, br.Memo, br.ChainID, br.Gas, br.GasAdjustment,
-		br.AccountNumber, br.Sequence, br.Fees, br.GasPrices, br.GenerateOnly, br.Simulate,
-	)
-}
-
 // ValidateBasic performs basic validation of a BaseReq. If custom validation
 // logic is needed, the implementing request handler should perform those
 // checks manually.
-func (br BaseReq) ValidateBasic(w http.ResponseWriter) bool {
+func ValidateBasic(w http.ResponseWriter, br client.BaseReq) bool {
```

Comment:
Why? This function is `BaseReq` specific. I'd suggest to either revert this or give the function a more appropriate name, e.g. `ValidateBaseReq`

Reply:
That was the original name. Given that the scope of this ticket was not related, I tried not to be too intrusive with respect to name. Specially given that it is used everywhere.
Also it was necessary to move it over to avoid circular dependencies.

Reply:
This is linked to #3487, I will wait for that PR to get merge to see if this is still applicable.